### PR TITLE
[BUG] allow from root

### PR DESCRIPTION
**Pull request by Aigerim-ai**
### Description
[Describe what this change achieves]
 Resolves running **yarn odd bootstrap** from root  with **--allow-root** flag
------------------
added changelog.md
### Issues Resolved
[List any issues this PR will resolve]
 https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3624
### Check List
- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [x] New functionality includes testing.
- [x] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by Aigerim-ai** [comment]
@ashwin-pc  done

**Comment by Aigerim-ai** [comment]
@abbyhu2000 Hello! Could you please approve this PR?

File: src/setup_node_env/root/force.js [hunk 1]
```diff
@@ -29,8 +29,12 @@
  */
 
 module.exports = function (argv) {
-  var rootIndex = argv.indexOf('--allow-root');
+  var args = process.argv.slice(2);
+  var rootIndex = args.indexOf('--allow-root');
   var force = rootIndex >= 0;
   if (force) argv.splice(rootIndex, 1);
```
Comment (ashwin-pc) [inline src/setup_node_env/root/force.js:35]: do we need this since we are already removing the flag in the next line?

**Comment by joshuarrrr** [comment]
@Aigerim-ai Check the new unit tests - it looks like a couple of the test cases aren't passing.

**Comment by joshuarrrr** [comment]
@Aigerim-ai I'm going to set this to draft until you're able to take a look at the unit test failures.

**Comment by Aigerim-ai** [comment]
Hi @joshuarrrr ! I have changed tests, now it is working

**Pull request closed without merging**